beacon-chain/state/stategen/getter_test.go

```diff
@@ -64,41 +62,58 @@ func TestStateByRoot_ColdState(t *testing.T) {
 	}
 }
 
-func TestStateByRoot_HotStateDB(t *testing.T) {
+func TestStateByRoot_HotStateUsingEpochBoundaryCacheNoReplay(t *testing.T) {
 	ctx := context.Background()
 	db, _ := testDB.SetupDB(t)
 
 	service := New(db, cache.NewStateSummaryCache())
 
 	beaconState, _ := testutil.DeterministicGenesisState(t, 32)
+	if err := beaconState.SetSlot(10); err != nil {
+		t.Fatal(err)
+	}
 	blk := &ethpb.SignedBeaconBlock{Block: &ethpb.BeaconBlock{}}
 	blkRoot, err := stateutil.BlockRoot(blk.Block)
 	if err != nil {
 		t.Fatal(err)
 	}
-	if err := service.beaconDB.SaveBlock(ctx, blk); err != nil {
+	if err := service.beaconDB.SaveStateSummary(ctx, &pb.StateSummary{Root: blkRoot[:]}); err != nil {
 		t.Fatal(err)
 	}
-	if err := service.beaconDB.SaveGenesisBlockRoot(ctx, blkRoot); err != nil {
+	if err := service.epochBoundaryStateCache.put(blkRoot, beaconState); err != nil {
 		t.Fatal(err)
 	}
-	if err := service.beaconDB.SaveState(ctx, beaconState, blkRoot); err != nil {
+	loadedState, err := service.StateByRoot(ctx, blkRoot)
+	if err != nil {
 		t.Fatal(err)
 	}
-	targetSlot := uint64(10)
-	targetBlock := &ethpb.SignedBeaconBlock{Block: &ethpb.BeaconBlock{Slot: targetSlot, ParentRoot: blkRoot[:], ProposerIndex: 7}}
-	if err := service.beaconDB.SaveBlock(ctx, targetBlock); err != nil {
-		t.Fatal(err)
+	if loadedState.Slot() != 10 {
+		t.Error("Did not correctly load state")
 	}
-	targetRoot, err := stateutil.BlockRoot(targetBlock.Block)
+}
+
+func TestStateByRoot_HotStateUsingEpochBoundaryCacheWithReplay(t *testing.T) {
+	ctx := context.Background()
+	db, _ := testDB.SetupDB(t)
+
+	service := New(db, cache.NewStateSummaryCache())
```

Comment:
It's probably better to use the same state summary cache as the db for these tests.

```suggestion
	db, ssc := testDB.SetupDB(t)

	service := New(db, ssc)
```